harvdev_utils/psycopg_functions/fb_feature_classes.py

```diff
@@ -372,27 +371,19 @@ def classify_allele(self):
         return
 
     def determine_gene_action(self):
-        """Determine if transgenic allele expresses, targets, or is_regulated_by its gene."""
+        """Determine if transgenic allele expresses or targets its parent gene."""
         self.classify_allele()
-        if self.classification == 'transgenic':
-            if type(self.crispr_ko_coll) != list:
-                log.warning('Allele {} needs "crispr_ko_coll" to determine "gene_action".'.format(self.uniquename))
-            elif len(self.crispr_ko_coll) > 0:
-                self.gene_action = 'targets'
-                self.targets = self.gene_id
-            elif type(self.mut_origin) != list:
-                log.warning('Allele {} needs "mut_origin" CV terms to determine "gene_action".'.format(self.uniquename))
-            else:
-                cvterm_counter = 0
-                for term in self.mut_origin:
-                    if term in ['in vitro construct - RNAi']:
-                        cvterm_counter += 1
-                if cvterm_counter > 0:
-                    self.gene_action = 'targets'
-                    self.targets = self.gene_id
-                else:
-                    self.gene_action = 'expresses'
-                    self.expresses = self.gene_id
+        if self.classification != 'transgenic':
+            return
+        if self.transgenic_class is None:
+            log.warning(f'Allele {self.name} ({self.uniquename}) is missing the transgenic class info needed to determine its action on the parent gene.')
+            return
+        if 'sgRNA' in self.transgenic_class or 'RNAi_reagent' in self.transgenic_class:
```

Comment:
I think you need to add 'antisense' to the list (sgRNA, RNAi_reagent) that turns self.gene-action into 'targets' (there are already alleles that have 'antisense' in transgenic_class and this would make it match Peeves checking which classes antisense as a targeting type along wiwth sgRNA and RNAi_reagent).